spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/support/DatastoreRepositoryFactory.java

```diff
@@ -118,21 +119,34 @@ public void setApplicationContext(ApplicationContext applicationContext)
 
 	private QueryMethodEvaluationContextProvider delegateContextProvider(
 			QueryMethodEvaluationContextProvider evaluationContextProvider) {
+
 		return new QueryMethodEvaluationContextProvider() {
 			@Override
 			public <T extends Parameters<?, ?>> EvaluationContext getEvaluationContext(
 					T parameters, Object[] parameterValues) {
 				StandardEvaluationContext evaluationContext = (StandardEvaluationContext)
-						evaluationContextProvider
-						.getEvaluationContext(parameters, parameterValues);
+						evaluationContextProvider.getEvaluationContext(parameters, parameterValues);
 				evaluationContext.setRootObject(
 						DatastoreRepositoryFactory.this.applicationContext);
 				evaluationContext.addPropertyAccessor(new BeanFactoryAccessor());
 				evaluationContext.setBeanResolver(new BeanFactoryResolver(
 						DatastoreRepositoryFactory.this.applicationContext));
 				return evaluationContext;
 			}
+
+			@Override
+			public <T extends Parameters<?, ?>> EvaluationContext getEvaluationContext(
+					T parameters, Object[] parameterValues, ExpressionDependencies expressionDependencies) {
+				StandardEvaluationContext evaluationContext =
```

Comment:
How did you figure this specific implementation?

@mp911de Any suggestions on what the default implementation should be?

Reply:
I copied another class which implements the interface: https://github.com/spring-projects/spring-data-commons/blob/fb15a9647d9da3f4df20748b1b8de0570e4a2579/src/main/java/org/springframework/data/repository/query/ExtensionAwareQueryMethodEvaluationContextProvider.java#L93

It looks like the only extra step is to add the `ExpressionDependencies` as an extra input when creating the context.

Reply:
`ExpressionDependencies` is an optimization to use only `EvaluationContextExtension`s that are required by the SpEL expression to be evaluated. `ExpressionDependencies` was primarily driven by Reactive SpEL execution (see `ReactiveQueryMethodEvaluationContextProvider`).

On a related note: Why do you have this delegate in place? Or asked differently, what doesn't work if you use the `QueryMethodEvaluationContextProvider` provided by `RepositoryFactoryBeanSupport`?

Reply:
> On a related note: Why do you have this delegate in place? Or asked differently, what doesn't work if you use the QueryMethodEvaluationContextProvider provided by RepositoryFactoryBeanSupport?

Hmm good question, I'm not sure - this code was written a while ago and am not sure of the context behind it.

We can definitely experiment to see if it can be removed, would be nice to reduce some code complexity.